packages/headless/src/modules/super-admin/service/index.ts

```diff
@@ -1,59 +1,295 @@
-        }
-        if(!body?.password){
-            return errorResponse('Password required!', null, HttpStatus.BAD_REQUEST)
-        }
-        
-        body.password = await bcrypt.hash(body.password, authConfig.salt)
-
-        const newSuperAdmin = await this.superAdminRepository.create(body)
-        if(newSuperAdmin){
-            return successResponse(PartialType(SuperAdmin), newSuperAdmin)
-        }
-        return errorResponse('Error in create super admin', null, HttpStatus.CONFLICT)
-    }
-
-    async superAdminLogin(body: SuperAdminLoginDto): Promise<IServiceResponse<SuperAdminLoginRes>>{
-        const userData = await this.superAdminRepository.findOne({email: body.email})
-        if(!userData){
-            return errorResponse('Invalid credentials!', null, HttpStatus.BAD_REQUEST)
-        }
-
-        const matchPassword = await bcrypt.compare(body.password, userData.password)
-        if(!matchPassword){
-            return errorResponse('Invalid credentials!', null, HttpStatus.BAD_REQUEST)
-        }
-
-        const payload: AdminJwtPayload = {
-            id: userData.id,
-            username: userData.firstName+' '+userData.lastName,
-            logInTime: Date.now(),
-            role: userData.role
-        }
-        const token = this.jwtService.sign(payload)
-        return successResponse(SuperAdminLoginRes, {token})
-    }
-}
\ No newline at end of file
+export class SuperAdminService {
+  constructor(
+    private readonly superAdminRepository: SuperAdminRepository,
+    private readonly superAdminHelperService: SuperAdminHelperService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async getProfileData(
+    id: string,
+  ): Promise<IServiceResponse<Partial<SuperAdmin>>> {
+    const profileData = await this.superAdminRepository.findOne({ id });
+    if (profileData) {
+      delete profileData.password;
+      delete profileData.isMfaEnabled;
+      delete profileData.mfaType;
+      return successResponse(SuperAdmin, profileData);
+    }
+    return errorResponse('User not found', null, HttpStatus.NOT_FOUND);
+  }
+  async superAdminCreate(
+    body: SuperAdminSignupReq,
+  ): Promise<IServiceResponse<Partial<SuperAdmin>>> {
+    let isExist = await this.superAdminRepository.findOne({
+      email: body.email,
+    });
+    if (isExist?.email === body.email) {
+      return errorResponse(
+        'This email is already exist.',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    if (body?.phone) {
+      isExist = await this.superAdminRepository.findOne({ phone: body.phone });
+      if (isExist?.phone === body.phone) {
+        return errorResponse(
+          'This phone number is already exist.',
+          null,
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+    }
+    if (!body?.password) {
+      return errorResponse('Password required!', null, HttpStatus.BAD_REQUEST);
```

Comment:
why we are validating here instead of validating by class-validator? i think this validation can be done by class-validator. correct me if i am wrong